features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Overview.jsx

```diff
@@ -183,59 +266,17 @@ function Overview() {
                     setSwaggerDescription('');
                 });
         } else {
-            restApi.getDocumentsByAPIId(api.id)
-                .then((response) => {
-                    const overviewDoc = response.body.list.filter((item) => item.otherTypeName === '_overview');
-                    if (overviewDoc.length > 0) {
-                        // We can override the UI with this content
-                        setOverviewDocOverride(overviewDoc[0]); // Only one doc we can render
-                    }
-                    setAllDocuments(response.body.list);
-                })
-                .catch((error) => {
-                    if (process.env.NODE_ENV !== 'production') {
-                        console.log(error);
-                    }
-                    const { status } = error;
-                    if (status === 404) {
-                        Alert.error(intl.formatMessage({
-                            id: 'Apis.Details.Overview.error.occurred',
-                            defaultMessage: 'Error occurred',
-                        }));
-                        setNotFound(true);
-                    }
-                    setAllDocuments([]);
-                });
-            restApi.getAllTiers('subscription')
-                .then((response) => {
-                    try {
-                        // Filter policies base on async or not.
-                        const filteredList = response.body.list.filter((str) => isApiPolicy(str.name));
-                        setAllPolicies(filteredList);
-                    } catch (e) {
-                        console.log(e);
-                        Alert.error(intl.formatMessage({
-                            id: 'Apis.Details.Overview.error.occurred',
-                            defaultMessage: 'Error occurred',
-                        }));
-                    }
-                }).catch((error) => {
-                    if (process.env.NODE_ENV !== 'production') {
-                        console.log(error);
-                    }
-                    const { status } = error;
-                    if (status === 404) {
-                        Alert.error(intl.formatMessage({
-                            id: 'Apis.Details.Overview.error.occurred',
-                            defaultMessage: 'Error occurred',
-                        }));
-                        setNotFound(true);
-                    }
-                    setAllDocuments([]);
+            setIsLoading(true);
+            Promise.all([getDocuments(), getSubscriptionPolicies()])
+                .then(() => {
+                    setIsLoading(false);
+                }).catch(() => {
+                    setIsLoading(false);
```

Comment:
Same as above , please log and show an error message to the user

Reply:
same explanation as above https://github.com/wso2/carbon-apimgt/pull/10298#discussion_r610555829